client/asset/btc/spv.go

```diff
@@ -611,44 +658,20 @@ func (w *spvWallet) listUnspent() ([]*ListUnspentResult, error) {
 	res := make([]*ListUnspentResult, 0, len(unspents))
 	for _, utxo := range unspents {
 		// If the utxo is unconfirmed, we should determine whether it's "safe"
-		// by seeing if we control the inputs of its transaction.
-		var safe bool
-		if utxo.Confirmations > 0 {
-			safe = true
-		} else {
-			txHash, err := chainhash.NewHashFromStr(utxo.TxID)
-			if err != nil {
-				return nil, fmt.Errorf("error decoding txid %q: %v", utxo.TxID, err)
-			}
-			txDetails, err := w.wallet.walletTransaction(txHash)
-			if err != nil {
-				return nil, fmt.Errorf("walletTransaction error: %v", err)
-			}
-			// To be "safe", we need to show that we own the inputs for the
-			// utxo's transaction. We'll just try to find one.
-			safe = true
-			// TODO: Keep a cache of our redemption outputs and allow those as
-			// safe inputs.
-			for _, txIn := range txDetails.MsgTx.TxIn {
-				_, _, _, _, err := w.wallet.FetchInputInfo(&txIn.PreviousOutPoint)
-				if err != nil {
-					if !errors.Is(err, wallet.ErrNotMine) {
-						w.log.Warnf("FetchInputInfo error: %v", err)
-					}
-					safe = false
-					break
-				}
-			}
-		}
+		// by seeing if we control the inputs of its transaction. TODO: Keep a
+		// cache of our redemption outputs and allow those as safe inputs.
```

Comment:
I meant to change this comment.  The TODO here is debatable.  The contract output being spent in the redeem is arguably untrusted.  The other party might still have a claim on it via redeem.  It all depends on network conditions and the lock time on the contract.  I think I'd rather play it safe with these.